File: src/main/java/misc/UnmodifiableCollection.java
package misc;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

public class UnmodifiableCollection {

    public static void main(String[] args) {

        List<Integer> list = new ArrayList<>();

        list.add(1);
        list.add(2);
        list.add(3);

        Collection<Integer> unmodifiableCollection = Collections.unmodifiableCollection(list);

//        unmodifiableCollection.add(4); // throws UnsupportedOperationException

    }

}

File: src/main/java/misc/UnmodifiableCollection.kt
package misc

import java.util.*

object UnmodifiableCollection {

    @JvmStatic fun main(args: Array<String>) {

        val list = ArrayList<Int>()

        list.add(1)
        list.add(2)
        list.add(3)

        @SuppressWarnings
        val unmodifiableCollection = Collections.unmodifiableCollection(list)

        //        unmodifiableCollection.add(4); // throws UnsupportedOperationException

    }

}
